### PR TITLE
[FIX][l10n_it_edi] remove auto_install

### PR DESCRIPTION
**Pull request by dcorio**
This module isn't ready yet and therefore should not be set as auto_install.

We would love to have an official Italian localization but at the moment we have to rely on the OCA modules to create, send and receive electronic invoices.
When a customer creates a new database and installs the Italian chart of accounts, l10n_it_edi gets automatically installed and introduces the following problems:

1. Customers use the fields created by this modules without knowing that they'll soon have to uninstall it
2. We always have to manually export/import the data in order to migrate the customer from l10n_it_edi to OCA modules
3. Customers get to us quite frustrated because Odoo's direct team told them that Odoo could manage Italian e-invoicing
4. We can't use any CI/CD system (Travis, Runbot, etc) because we'd need to manually uninstall l10n_it_edi as it's not compatible with the OCA's italian localization

PS: I'm well aware that you won't accept this PR as I've already opened a ticket using an Odoo Enterprise license code and you already refused to modify this module.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by scigghia** [comment]
👍 I confirm and endorse what Davide has written

**Comment by GSLabIt** [comment]
go for it 👍🏼

**Comment by CarFla** [comment]
+1 I confirm and endorse what Davide has written

**Comment by iw3hxn** [comment]
+1 confirm the issue open from Davide

**Comment by jado95** [comment]
i agree  👍🏼

**Comment by alessandrocamilli** [comment]
+1

**Comment by Byloth** [comment]
👏👏👏

**Review by sergiocorato — APPROVED** [review]
+1

**Review by eLBati — APPROVED** [review]
I confirm `l10n_it_edi` is not usable in Italy and should not be automatically installed

**Review by scigghia — APPROVED** [review]

**Comment by primes2h** [comment]
I entirely agree with every word Davide said.

In fact it's even worse than that, `auto_install': True` option on `l10n_it_edi` brings the following modules to be auto-installed too: 
`account_edi`
`account_edi_facturx`
`account_edi_ubl`

**Review by primes2h — APPROVED** [review]

**Review by Byloth — APPROVED** [review]

**Review by GSLabIt — APPROVED** [review]

**Review by alessandrocamilli — APPROVED** [review]

**Comment by AIR-Leo** [comment]
Totally agree 👍🏼

**Comment by OpenCode** [comment]
+1

**Review by robyf70 — APPROVED** [review]
Please APPROVE ASAP!

**Review by marcelofrare — APPROVED** [review]
I agree with @dcorio

**Review by andreampiovesana — APPROVED** [review]
I agree with @dcorio

**Comment by SGiuseppe** [comment]
+1

**Comment by dcorio** [comment]
> I entirely agree with every word Davide said. 👍
> 
> In fact it's even worse than that, `'auto_install': True` option on `l10n_it_edi` brings the following modules to be auto-installed too:
> `account_edi`
> `account_edi_facturx`
> `account_edi_ubl`

Correct, that messes up things even more and creates a lot of confusion

**Comment by elvise** [comment]
+1

**Review by jado95 — APPROVED** [review]

**Review by mymage — APPROVED** [review]

**Comment by PatosMarco** [comment]
+1

**Comment by stevech091** [comment]
+1 with @dcorio

**Review by MarcoCalcagni — APPROVED** [review]
It is Urgent
Please approve the PR

**Comment by OmniaGit** [comment]
+1 @dcorio

**Comment by jco-odoo** [comment]
@dcorio You are making things really difficult...  I thought you were going to simplify the export in the OCA module as well, such that we can have a common QWeb, but I have not seen anything yet (maybe did not look at the right place...).  I see there is good willingness, talking e.g. to Andrea, ... but this is not constructive in any way.

**Comment by william-andre** [comment]
+1 @jco-odoo

**Comment by Byloth** [comment]
image

**Comment by mrcast** [comment]
I agree with @dcorio

**Comment by william-andre** [comment]
Spam +1

**Comment by dcorio** [comment]
@jco-odoo "not constructive"?
I'm just asking to NON install by default a "NOT" working (yet) module.

That module isn't ready yet. It doesn't even cover the most basic cases needed by the most basic companies.

We'd really hope to see, one day, only one Italian localization managed directly by Odoo, but that won't happen really soon.
The Italian Community is definitely open to collaborations as long as the code remains available on Odoo CE of course...

Right now, you're just making things harder for those who need to implement Odoo in Italy for real.

And keep in mind that I'm an official partner that DOES NOT SELL Odoo Community.
All the Odoo installation I manage are based on Odoo Enterprise on Odoo.sh.

**Comment by jco-odoo** [comment]
This module is in community and you did nothing...  Not even basic arguments why this module could not be integrated in the OCA version as we don't need to agree on everything, but a common export/import of the format could be a nice minimum e.g.  

We will resurrect anyways...

**Comment by sergiocorato** [comment]
@jco-odoo I agree that a nice minimum would be a good idea, but as in Italy we have a full working solution from more than 2 years, a project manager who wants to 1. achieve the result 2. do not waste money, would simply COPY the logic of the basic of the working solution.
Asking for help if needed or proposing improvements (as we do normally in OCA).

**Comment by jco-odoo** [comment]
@sergiocorato A project manager wants something that follows the standard as well.

**Comment by zeroincombenze** [comment]
May be argue Davide was rude but I am afraid his matter is the same of all Italians developers. In my opinion you should listen at Italian people who is building the Odoo successful.
Do you do not believe this module is always uninstalled by Italian users? Do you do not trust in Italian developers?

**Comment by sergiocorato** [comment]
@jco-odoo Our code could not be the best and can surely be improved, as it comes from a solution developed to be used in time to fulfill legal requirements in Italy, by some company or professionals or even single person with nothing todo with a big company as Odoo.
Anyway Odoo itself wasn't able to release a solution for years, so I'm not sure the project manager did a good job until now.

**Comment by william-andre** [comment]
I am reading this from an external point of view as I was not part of previous conversations. But this thread really does not look like you want to have a constructive discussion. At all.
Why is it not better to fix the module instead of disabling it? Provide more information, or links to relevant discussions. 

____________________

> This module isn't ready yet and therefore should not be set as auto_install.

Why is it not ready? What is not working?


> We would love to have an official Italian localization but at the moment we have to rely on the OCA modules to create, send and receive electronic invoices.

Then propose a fix in the official modules?

> When a customer creates a new database and installs the Italian chart of accounts, l10n_it_edi gets automatically installed and introduces the following problems:
>  1. Customers use the fields created by this modules without knowing that they'll soon have to uninstall it
>  2. We always have to manually export/import the data in order to migrate the customer from l10n_it_edi to OCA modules

Those two points are the same, really...

>  3. Customers get to us quite frustrated because Odoo's direct team told them that Odoo could manage Italian e-invoicing

We understand, but it is not a reason not to fix the module Odoo's direct team said would work instead of forcing another one, that is not available on the SaaS.

>  4. We can't use any CI/CD system (Travis, Runbot, etc) because we'd need to manually uninstall l10n_it_edi as it's not compatible with the OCA's italian localization

Why would the CI/CD fail? It sure doesn't fail on our end. Runbot v14: https://runbot.odoo.com/runbot/build/6562171

> PS: I'm well aware that you won't accept this PR as I've already opened a ticket using an Odoo Enterprise license code and you already refused to modify this module.

Why spam then? This whole thread seems like an attempt of bomb trying to make a mess.

______________

> I entirely agree with every word Davide said. +1
> 
> In fact it's even worse than that, `'auto_install': True` option on `l10n_it_edi` brings the following modules to be auto-installed too:
> `account_edi`
> `account_edi_facturx`
> `account_edi_ubl`

That is simply not true, even completely wrong. `auto_install` will install the module automatically IF all the dependence are already installed. This is how all bridge modules all work (`account_fleet`, `pos_hr`, `sale_stock`, ... **and** `l10n_it_edi`)
___________________

> Anyway Odoo itself wasn't able to release a solution for years, so I'm not sure the project manager did a good job until now.

Then provide us the information needed to fix that. This thread looks more like an angry Twitter thread than anything else.

**Comment by jco-odoo** [comment]
@zeroincombenze The module is usable, which does not say it can be more usable.  I trust that the solutions developed by OCA and the Italian developers fulfill their purpose and I am sure there are some very good features in it.  But the point is that developing something as a solution for clients or developing something for Odoo standard (and keeping and maintaining that standard) is something entirely different.  And we can use all the help.  

@sergiocorato To an extent, we could agree with the 'until now' part, but let's build for the future.

**Comment by GSLabIt** [comment]
> @zeroincombenze The module is usable, which does not say it can be more usable.  I trust that the solutions developed by OCA and the Italian developers fulfill their purpose and I am sure there are some very good features in it.  But the point is that developing something as a solution for clients or developing something for Odoo standard (and keeping and maintaining that standard) is something entirely different.  And we can use all the help.  
> 
> @sergiocorato To an extent, we could agree with the 'until now' part, but let's build for the future.  

Did u manage to solve this:
https://github.com/odoo/odoo/issues/41511

**Comment by dcorio** [comment]
> Why spam then? This whole thread seems like an attempt of bomb trying to make a mess.

@jco-odoo it's not spam, it's a desperate attempt to make Odoo understand that we the partners already know what customers need. Probably better than a Belgian company if you will...
It took 10 years of work to get where we are now.

We're not saying that what you implemented so far is crap and must be thrown away.
The problem is that we need a fully complete accounting package for v14 and re-developing what we have in order to be based on l10n_it_edi would require probably a year.

So, for the moment, I cannot see the possibility to integrate the OCA's italian localization into l10n_it_edi.
And that would make no sense anyway IMO.

We want to contribute to l10n_it_edi and we will.
The only problem is that we are facing angry customers in desperate need of a solution and having l10n_it_edi auto installed is a technical problem.

We're just speaking about that "auto_install". That's it :)

I really hope that one day we, Odoo and the Italian partners, could sit at a table together and discuss how to improve Odoo's Italian localization.
And that moment could even be in just a few weeks, but right now we just need a fully working accounting package for Odoo 14 and let the customers know that Odoo 14 can manage the e-invoicing in Italy for real, thus we don't have the time for that.

So @jco-odoo, please don't take this personal, because it's not.
You've been charged with a task as awful as the Italian e-invoicing!
I admire your bravery! ;-)

**To everybody: we can now calm down and avoid posting any more comments. The goal has been reached.**

**Comment by primes2h** [comment]
> > I entirely agree with every word Davide said. +1
> > In fact it's even worse than that, `'auto_install': True` option on `l10n_it_edi` brings the following modules to be auto-installed too:
> > `account_edi`
> > `account_edi_facturx`
> > `account_edi_ubl`
> 
> That is simply not true, even completely wrong. `auto_install` will install the module automatically IF all the dependence are already installed. This is how all bridge modules all work (`account_fleet`, `pos_hr`, `sale_stock`, ... **and** `l10n_it_edi`)

@william-andre 

Let me put it in another way.

1) Create a new empty DB with only `base` module installed.
2) Install the italian standard CoA `l10n_it`

You'll end up with all this extra modules installed.
`l10n_it_edi`
 `account_edi`
`account_edi_facturx`
`account_edi_ubl`


Now redo 1) and 2) __but__ this time remove `'auto_install': True` in `l10_it_edi`.
No extra modules will be installed.

Did you get the point?

**Comment by andreampiovesana** [comment]
Why closed?

**Comment by jco-odoo** [comment]
@primes2h It is because account_edi is auto_install on account.  And you can uninstall the module...  You can even uninstall account_edi and then it will not install l10n_it_edi afterwards.  

@andreampiovesana Because we are not going to merge it.

**Comment by william-andre** [comment]
@eLBati I don't know if it really got no answer... I see answers on the pull request, then I see this too https://github.com/odoo/odoo/pulls?q=is%3Apr+ddt+
With this that got merged yesterday https://github.com/odoo/odoo/pull/65591 (I can assure to you that the fact that it got merged got nothing to do with this thread)

____________________

@primes2h Did you even try what you are saying? 

`odoo-bin -i account,l10n_it` with `auto_install: True`
```
14.0=> SELECT name FROM ir_module_module WHERE name LIKE '%edi%' AND state = 'installed';
        name         
---------------------
 account_edi
 account_edi_facturx
 account_edi_ubl
 l10n_it_edi
 web_editor
(5 rows)
```

`odoo-bin -i account,l10n_it` with `auto_install: False`
```
14.0=> SELECT name FROM ir_module_module WHERE name LIKE '%edi%' AND state = 'installed';
        name         
---------------------
 account_edi
 account_edi_facturx
 account_edi_ubl
 web_editor
(4 rows)
```

**Comment by primes2h** [comment]
@william-andre 
I've been inaccurate, I'm sorry (auto-install has be removed from `account_edi` too).

Anyway, what I pointed out [here](https://github.com/odoo/odoo/pull/67185#issuecomment-793566942) still stand, e-invoicing stuff should not be installed by default imho, least of all if it's not ready as it is the italian one now.
My 2 cents.

**Comment by william-andre** [comment]
`account_edi` is installed by default so that we don't have to rely on the OCR and image recognition to be able to send invoices from one Odoo instance to another, and retrieve the data with 100% accuracy. You will loose that feature if you remove the installation of that module.
By the way, that feature was already present since version 12.0

Clients on the SaaS expect the EDI to work too. You still did not explain why it doesn't work with the standard module.

**Comment by scigghia** [comment]
The Italian community has been looking for a stable solution regarding Italian accounting for many years.

In the meantime we have closed the gap under OCA and we all would like to see Odoo s.a. ready to listen the suggestions from those who make possible the + 94% of revenue in Italy

Today we could solve the problem with a simple flag, tomorrow maybe not.
Currently, our community (made of a large group of consultants and companies who believe and invest in Odoo EE / CE) is sending out a significant number of signs and we all need to catch.

Despite several tentative done both personally and by other members, the perception as been only lack of dialog and openess.

I understand somebody may have passed the line with regards of communication style, but it is a consequence of people that put much passion into it and strongly believe in Odoo.

As also I have been able to explain you, i'm committed to help members understand role and responsabilties, on the other hand we all want to achieve a common goal that is represented by "Odoo as the best market choice" even in Italy, and to do so we must continue to develop without forgetting above all to protect customers, who are the first source of energy.

Bottom line, we would save our common time.

Regards
**Andrea Cometa - Odoo Italia Association President**

**Comment by claudiostevanatocylix** [comment]
Today we had a problem with an installation for a new customer, caused by this auto_install in the module l10n_it_edi, unused by 99.99% of italians Odoo developers
